features/other_a/gift.feature

```diff
@@ -337,3 +337,26 @@ Feature: Create Gifts
       And I should not see "by gifter for giftee1"
     When I view the work "Rude Gift"
     Then I should not see "For giftee1."
+
+  Scenario: Can't give a gift to a user who has blocked you
+    Given the user "giftee1" has blocked the user "gifter"
+    When I am logged in as "gifter"
+      And I post the work "Rude Gift" as a gift for "giftee1"
+    Then I should see "Sorry! We couldn't save this work because: giftee1 does not accept gifts from you."
+      And 0 emails should be delivered to "giftee1@example.com"
+
+  Scenario: Can't gift an existing work to a user who has blocked you
+    Given the user "giftee1" has blocked the user "gifter"
+      And I press "Post"
+      And I follow "Edit"
+      And I give the work to "giftee1"
+    When I press "Post"
+    Then I should see "Sorry! We couldn't save this work because: giftee1 does not accept gifts from you."
+
+  Scenario: Can't gift a work whose co-creator is blocked by recipient
+    Given I coauthored the work "Collateral" as "gifter" with "gifter2"
+      And the user "giftee1" has blocked the user "gifter2"
+      And I edit the work "Collateral"
+      And I give the work to "giftee1"
+    When I press "Post"
+    Then I should see "Sorry! We couldn't save this work because: giftee1 does not accept gifts from gifter2."
```

Comment:
I don't think this should be revealed according to the Jira spec.

Reply:
Fixed, thanks.

I'm also wondering if the "from you" is a good idea or if it wouldn't be nicer to keep quiet about that fact and also display the standard message. Would make for simpler code too.